support-frontend/app/controllers/DigitalSubscriptionController.scala

```diff
@@ -51,24 +55,29 @@ class DigitalSubscriptionController(
 
   def digital(countryCode: String, orderIsAGift: Boolean): Action[AnyContent] = CachedAction() { implicit request =>
     implicit val settings: AllSettings = settingsProvider.getAllSettings()
-    val title = if (orderIsAGift) {
-      "Support the Guardian | The Guardian Digital Gift Subscription"
-     } else {
-       "Support the Guardian | The Guardian Digital Subscription"
-     }
+    val title = s"Support the Guardian | The Guardian Digital ${if (orderIsAGift) "Gift " else ""}Subscription"
     val mainElement = EmptyDiv("digital-subscription-landing-page-" + countryCode)
     val js = Left(RefPath("digitalSubscriptionLandingPage.js"))
     val css = Left(RefPath("digitalSubscriptionLandingPage.css"))
     val description = stringsConfig.digitalPackLandingDescription
     val canonicalLink = Some(buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift))
-    val promoCodes: List[PromoCode] = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil) ++ DefaultPromotions.DigitalSubscription.all
+    val queryPromos = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
+    val promoCodes: List[PromoCode] = queryPromos ++ DefaultPromotions.DigitalSubscription.all
     val hrefLangLinks = Map(
       "en-us" -> buildCanonicalDigitalSubscriptionLink("us", orderIsAGift),
       "en-gb" -> buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift),
       "en-au" -> buildCanonicalDigitalSubscriptionLink("au", orderIsAGift),
       "en" -> buildCanonicalDigitalSubscriptionLink("int", orderIsAGift)
     )
-
+    val country = (for {
+      countryGroup <- CountryGroup.byId(countryCode)
+      country <- countryGroup.countries.headOption
+    } yield country).getOrElse(UK)
+    val maybePromotionCopy = queryPromos.headOption.flatMap(promoCode =>
+      ProductPromotionCopy(promotionServiceProvider
+        .forUser(false), stage)
+        .getCopyForPromoCode(promoCode, DigitalPack, country)
+    )
```

Comment:
just indentation/formatting
```suggestion
    val maybePromotionCopy = queryPromos.headOption.flatMap(promoCode =>
      ProductPromotionCopy(
        promotionServiceProvider.forUser(false),
        stage
      ).getCopyForPromoCode(promoCode, DigitalPack, country)
    )
```

Reply:
It looks like scalastyle doesn't like this as the method is now > 50 lines long (weirdly IntelliJ decided not to highlight this when I checked the change there 🤔). Not sure what the best thing to do is.

Reply:
Iv'e added another suggestion above in another comment!

I find intellij is a bit random on the scalastyle - mine seems to find wrnings in a lot of (other peoples) code that clearly passes teamcity, but other times teamcity starts complaining about things I've done that were not underlined.  I suspect there are different setting for different projects but haven't looked into it.